cellmlmanip/_singularity_fixes.py

```diff
@@ -0,0 +1,234 @@
+"""
+`remove_fixable_singularities` specifies a method to remove fixable singularities from a model
+This represents a flattened CellML model and metadata about its variables.
+"""
+from math import isclose
+
+import networkx as nx
+from sympy import (
+    Abs,
+    Add,
+    Eq,
+    Float,
+    I,
+    Mul,
+    Piecewise,
+    Pow,
+    Wild,
+    exp,
+    log,
+    solveset,
+)
+
+import cellmlmanip
+from cellmlmanip.model import FLOAT_PRECISION
+
+
+ONE = cellmlmanip.Quantity(1.0, 'dimensionless')
+
+
+def _generate_piecewise(vs, ve, sp, ex, V):
+    """Generates a piecewise for expression based on the singularity point (sp) and vmin (vs) and vmax (ve) """
```

Comment:
Something more like "Generates a new (piecewise) expression based on ``ex`` that returns a linear interpolation when variable ``V`` has a value between...

You could use a code block:
```
The returned expression between .. and .. is::

  f(vs) + (f(ve) - f(vs)) * (V - vs) / (ve - vs)

whenever::

  ...
  
``

Or even :math: if we're planning to allow latex in our docstrings at some point